aspnetcore/web-api/action-return-types.md

```diff
@@ -10,6 +10,212 @@ uid: web-api/action-return-types
 ---
 # Controller action return types in ASP.NET Core web API
 
+:::moniker range=">= aspnetcore-7.0"
+
+[View or download sample code](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/mvc/action-return-types/7.x/WebApiSample) ([how to download](xref:index#how-to-download-a-sample))
+
+ASP.NET Core provides the following options for web API controller action return types:
+
+* [Specific type](#specific-type)
+* [IActionResult](#iactionresult-type)
+* [ActionResult\<T>](#actionresultt-type)
+* [HttpResults](#httpresults-type)
+
+This article explains when it's most appropriate to use each return type.
+
+## Specific type
+
+The most basic action returns a primitive or complex data type, for example, `string` or a custom object. Consider the following action, which returns a collection of custom `Product` objects:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ProductsController.cs" id="snippet_Get":::
+
+Without known conditions to safeguard against, returning a specific type could suffice. The preceding action accepts no parameters, so parameter constraints validation isn't needed.
+
+When multiple return types are possible, it's common to mix an <xref:Microsoft.AspNetCore.Mvc.ActionResult> return type with the primitive or complex return type. Either [IActionResult](#iactionresult-type) or [ActionResult\<T>](#actionresultt-type) are necessary to accommodate this type of action. Several samples of multiple return types are provided in this article.
+
+### Return IEnumerable\<T> or IAsyncEnumerable\<T>
```

Comment:
Should we S&M this PR and I'll open an issue on this doc?
`### Return IEnumerable\<T> or IAsyncEnumerable\<T>` never answers the question or provides guidance.

Reply:
We do have in the best practices docs https://learn.microsoft.com/en-us/aspnet/core/performance/performance-best-practices?view=aspnetcore-6.0#return-ienumerablet-or-iasyncenumerablet but sure let's open an issue so we can improve it.

Reply:
@brunolins16  see #27210